Core/GDCore/IDE/Project/ProjectResourcesAdder.cpp

```diff
@@ -42,15 +42,13 @@ std::vector<gd::String> ProjectResourcesAdder::GetAllUseless(
   std::set<gd::String>& usedResources = resourcesInUse.GetAll(resourceType);
 
   // Search all resources not used
-  std::vector<gd::String> resources =
-      project.GetResourcesManager().GetAllResourceNames();
+  std::vector<std::shared_ptr<Resource>> resources =
+      project.GetResourcesManager().GetAllResources();
```

Comment:
nit: this works but also create a copy of the array of (smart) pointers into `resources`. While you only care about getting a... reference to it ;) 

Hence this should work:
```suggestion
  const std::vector<std::shared_ptr<Resource>>& resources =
      project.GetResourcesManager().GetAllResources();
```

(could also be written as `const auto& resources = ...` - which means "give me the resources vector, but just a reference to it. I won't even change it I swear.")

Reply:
Note that you need another change in the return value of `GetAllResources`, see my other comment.